clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -1030,6 +1030,7 @@ public void assign(Collection<TopicPartition> partitions) {
      * @return The fetched records (may be empty)
      */
     private Map<TopicPartition, List<ConsumerRecord<K, V>>> pollOnce(long timeout) {
+        client.poll(0L);
```

Comment:
@hachikuji 

> I wonder if we could just expose maybeTriggerWakeup() like you had a in a previous iteration?

That would definitely be faster here I think, plus it's a lot nicer with passing in the parameter for `poll` now since we save one `synchronize`. Should I expose it again?